src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -202,11 +205,61 @@ Image& Frame::image() const
     return *impl->m_image.get();
 }
 
+void Frame::clear_main_image()
+{
+    impl->m_image->clear(Color4f(0.0));
+}
+
 ImageStack& Frame::aov_images() const
 {
     return *impl->m_aov_images.get();
 }
 
+void Frame::add_aov(foundation::auto_release_ptr<AOV> aov)
+{
+    assert(aov.get());
+
+    size_t aov_index = aov_images().get_index(aov->get_name());
+    if (aov_index == size_t(~0) && aov_images().size() < MaxAOVCount)
+    {
+        aov_images().append(
+            aov->get_name(),
+            4, // aov->get_channel_count(),
+            PixelFormatFloat);
+        aovs().insert(aov);
+    }
+    else
+    {
+        RENDERER_LOG_WARNING(
+            "could not create %s AOV, maximum number of AOVs (" FMT_SIZE_T ") reached.",
```

Comment:
I know it's weird but all our messages are in lower case, so for the sake of consistency: *AOV* -> *aov*